sdk/python/feast/registry.py

```diff
@@ -356,57 +391,42 @@ def delete_feature_view(self, name: str, project: str):
             name: Name of feature view
             project: Feast project that this feature view belongs to
         """
+        found_feature_view = False
 
-        def updater(registry_proto: RegistryProto):
-            for idx, existing_feature_view_proto in enumerate(
-                registry_proto.feature_views
+        for idx, existing_feature_view_proto in enumerate(
+            self.cached_registry_proto.feature_views
+        ):
+            if (
+                existing_feature_view_proto.spec.name == name
+                and existing_feature_view_proto.spec.project == project
             ):
-                if (
-                    existing_feature_view_proto.spec.name == name
-                    and existing_feature_view_proto.spec.project == project
-                ):
-                    del registry_proto.feature_views[idx]
-                    return registry_proto
+                found_feature_view = True
+                del self.cached_registry_proto.feature_views[idx]
+
+        if not found_feature_view:
             raise FeatureViewNotFoundException(name, project)
 
-        self._registry_store.update_registry_proto(updater)
+        self._registry_store.update_registry_proto(self.cached_registry_proto)
 
     def refresh(self):
-        """Refreshes the state of the registry cache by fetching the registry state from the remote registry store."""
-        self._get_registry_proto(allow_cache=False)
+        """
+        Deprecated.
+        """
+        pass
```

Comment:
Should this maybe call `_get_registry_proto` to get the latest value from the registry store? Otherwise I think we don't have a way of getting the latest value after creating a registry object.

Reply:
Yea. How can I forcefully get the latest registry?

Reply:
There is an internal method for getting the latest registry, `_get_registry_proto`. This gets called on initialization, so the Registry object will be populated with the latest saved registry. Since the cached registry proto is always up-to-date, I don't see why `refresh` would ever be needed?